### DISCUSSION TITLE
Ubuntu 22.04/GNOME 41+42/Wayland

### DISCUSSION THREAD

**Discussion by ccope** [PaperWM-community/PaperWM#20] [Q&A] [answered]
Hi, I'm trying to set this extension up for the first time. I disabled the Desktop Icons NG extension, checked out the next-release branch, ran the install.sh script, accepted the settings changes, and rebooted, but it's showing the following error in the Extensions manager:

`Expected type string for argument 'property_name' but got type GObject_Object`

Clicking the settings gear button shows me:
```
TypeError: settings is undefined

Stack trace:
  syncStringSetting@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/prefs.js:722:5
  createFileChooserButton@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/prefs.js:700:22
  createWorkspacePage@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/prefs.js:237:49
  SettingsWidget@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/prefs.js:199:29
  buildPrefsWidget@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/prefs.js:744:20
  _init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:219:40
  OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:128:33
  asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:115:22
  run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
  main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  start@resource:///org/gnome/gjs/modules/script/package.js:190:8
  @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

Does anyone recognize this? How can I troubleshoot further?

**Comment by ccope** [comment]
Ah, I found this in my gnome-shell logs:
```
JS ERROR: Extension paperwm@hedning:matrix.org:
Error: Expected type string for argument 'property_name' but got type GObject_Object
notify@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/extension.js:197:12
errorNotification@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/extension.js:211:32
initUserConfig@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/extension.js:173:30
init@/home/cam/.local/share/gnome-shell/extensions/paperwm@hedning:matrix.org/extension.js:96:5
_callExtensionInit@resource:///org/gnome/shell/ui/extensionSystem.js:434:50
_callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:145:18
_onEnabledExtensionsChanged/<@resource:///org/gnome/shell/ui/extensionSystem.js:497:35
_onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:497:14
createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:528:46
enableExtension@resource:///org/gnome/shell/ui/extensionSystem.js:194:29
EnableExtension@resource:///org/gnome/shell/ui/shellDBus.js:423:38
_handleMethodCall@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:310:38
_wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:387:34
```

**Reply by ccope** [reply]
It seems like the error was thrown during the first-time setup of the user configuration directory. Logging out and back in made the extension work, it just seemed permanently broken because I thought gnome-extensions disable/enable was doing functional.

**Reply by Luvr681** [reply]
I got the same problem: Error: Expected type string for argument 'property_name' but got type GObject_Object . But after restarting, the error did not disappear and the extension does not work. Can you suggest what could be the issue?

**Reply by hikhvar** [reply]
Here the error is permanent, even restarting doesn't help Can't enable the extension in gnome-extension menu due to ` Expected type string for argument 'property_name' but got type GObject_Object`. Any idea?
I use Xorg instead of wayland since I need the screen sharing heavily for my remote job. And screensharing with wayland/pipewire is still a shit show.

**Comment by ccope** [comment]
Ha, iterating on/debugging this is fun. I have to log out and log back in because you cannot just restart the gnome shell because it is the Wayland display server, and you can't reload extensions either. https://superuser.com/questions/1164174/how-can-i-restart-the-gnome-shell-on-wayland

**Reply by smichel17** [reply]
Yep. The [official docs](https://gjs.guide/extensions/development/creating.html) on this suggest doing extension development in an X session so that you can restart.

**Reply by ccope** [reply]
It sucks because the extension management tools (CLI and GUI) don't warn you about this at all, so you think you're restarting the extension but actually nothing is happening >_>

**Comment by ccope** [comment]
I'm finding that moving between windows often causes the previous or current column to switch to the last size (selected through `super + R`) that it had before I switched into the column. This happens both by clicking between windows or moving with `super + ,`/`super + .`. It also happens in the vertical dimension if I have absorbed a two windows into the same column. I am not sure if this is an issue specific to Ubuntu 22.04 or a general bug with the extension. Watching the logs with `journalctl /usr/bin/gnome-shell -f`, I see:
```
Feb 14 15:33:35 cam-laptop org.gnome.Shell.desktop[2164]: resize-handler-width 1447
Feb 14 15:33:35 cam-laptop org.gnome.Shell.desktop[2164]: LAYOUT
Feb 14 15:33:36 cam-laptop org.gnome.Shell.desktop[2164]: resize-handler-width 450
Feb 14 15:33:36 cam-laptop org.gnome.Shell.desktop[2164]: LAYOUT
```

(The first resize-handler-width is me changing the size, the second is the next log output when I switched columns)

**Reply by ghost** [reply]
I have the same problem when using `Super + R` to resize a Tilix window under Debian with Gnome 41 (wayland). As soon I open a new window it can happen, that the previous window resets back to it's original size. It also happens when the window is not visible and I come back to it.

But I seem to have found a workaround: When I use the mouse to resize that window it stays that way even after opening new windows of the same kind with `Super + Enter`, but then I see some flicker as if the Tilix-window is not using the complete height for a brief part of a second. If I have used both, the mouse and `Super + R` to resize the window, the flickering seems to be gone in most cases.

I experienced this behaviour mostly with Tilix windows for now.

**Reply by ccope** [reply]
If I release the super key at my desired size and then do one more `Super + R`, when I switch columns it returns to the size I actually wanted.

**Reply by smichel17** [reply]
I had the exact same issue for a few days, but then it unexpectedly fixed itself. Unclear if it was the result of running system updates, switching from Wayland to X (although switching back did not re-trigger the bug), or something else. I may have reinstalled the extension; I don't remember.

**Comment by ccope** [comment]
Ubuntu 22.04 has now updated to GNOME 42, so this extension is listed as incompatible again.

**Reply by faebser** [reply]
You can actually just add "42" to the metadata.json to make it run in GNOME 42, but there is an error on startup.

**Reply by ccope** [reply]
I've started work in https://github.com/PaperWM-community/PaperWM/pull/25

**Comment by faebser** [comment]
Did you make any progress?

I have come to realize that I rely on this extension to work, because the simplicity and speed of how you move through windows is just amazing, therefore I am really motivated to get this to run in GNOME 42 again.

**Comment by faebser** [comment]
According to https://superuser.com/questions/1164174/how-can-i-restart-the-gnome-shell-on-wayland and https://unix.stackexchange.com/questions/627015/how-does-one-run-gnome-shell-in-a-window-nested-session there should be a way to run the extension in a nested gnome session and restart that session.

There is even a video about it: https://unix.stackexchange.com/questions/627015/how-does-one-run-gnome-shell-in-a-window-nested-session

**Reply by ccope** [reply]
Oh awesome, thanks! Looks like I don't need the GNOME_SHELL_SLOWDOWN_FACTOR bit but this will make debugging way less painful

**Comment by ccope** [comment, marked as answer]
The latest progress is here: https://github.com/PaperWM-community/PaperWM/pull/25
